holdings_maintenance/pds4/pds4linkshelf.py

```
@@ -0,0 +1,1270 @@
+        keys = list(dirdict.keys())
+        keys.sort()
+        for key in keys:
+            logger.error('Missing link shelf file entry for', key)
+
+        keys = list(shelfdict.keys())
+        keys.sort()
+        for key in keys:
+            logger.error('Link shelf file entry found for missing file', key)
+
+    except (Exception, KeyboardInterrupt) as e:
+        logger.exception(e)
```

Ruff (0.14.5)

Redundant exception object included in `logging.exception` call

(TRY401)